### DISCUSSION TITLE
Using argoCD to deploy velero helm chart

### DISCUSSION THREAD

**Discussion by parberge** [vmware-tanzu/velero#4770] [Community support Q&A] [answered]
We use argoCD to deploy the velero helm chart.
The deployment is working during initial deployment.
It creates all the necessary objects (e.g `backupstoragelocations.velero.io`).

This is however not reflected in the argoCD UI (no object connected to the CRD):
<img width="1371" alt="image" src="https://user-images.githubusercontent.com/5441497/159721252-bcacf9e8-2f0e-4071-b2b3-68939ffc98cf.png">

But I can see the object in kubernetes:
```
❯ kubectl get backupstoragelocations.velero.io -n velero
NAME   PHASE       LAST VALIDATED   AGE    DEFAULT
xxx    Available   47s              6m5s   

❯ velero backup-location get
NAME   PROVIDER   BUCKET/PREFIX            PHASE       LAST VALIDATED                  ACCESS MODE   DEFAULT
xxx    gcp        xxx-xxx-test-velero   Available   2022-03-23 15:21:01 +0100 CET   ReadWrite   
```

This becomes a problem when we in the code (i.e helm values) changes the bucket to something. ArgoCD will identify that it's out of sync, sync successfully, but the object isn't updated. 

Does anybody has experience getting it to work properly with argoCD?

**Comment by DJW-00000** [comment, marked as answer]
We have no issues with this, on ArgoCD 2.2.4 and Velero Helm Chart 2.27.3

![image](https://user-images.githubusercontent.com/74189361/160360017-bd52dfdf-35e4-4bb6-8a44-50571ba1ed18.png)

**Reply by parberge** [reply]
> We have no issues with this, on ArgoCD 2.2.4 and Velero Helm Chart 2.27.3
> 
> ![image](https://user-images.githubusercontent.com/74189361/160360017-bd52dfdf-35e4-4bb6-8a44-50571ba1ed18.png)

Oh interesting 🤔 . 

```
{
    "Version": "v2.1.6+a346cf9",
    "BuildDate": "2021-10-28T19:59:40Z",
    "GitCommit": "a346cf933e10d872eae26bff8e58c5e7ac40db25",
    "GitTreeState": "clean",
    "GoVersion": "go1.16.5",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v4.2.0 2021-06-30T22:49:26Z",
    "HelmVersion": "v3.6.0+g7f2df64",
    "KubectlVersion": "v0.21.0",
    "JsonnetVersion": "v0.17.0"
}
```
Is what we use. Can you share you version of all the stuff (By clicking the mascot in the upper left corner)

**Reply by DJW-00000** [reply]
```
 {
    "Version": "v2.2.4+78d749e",
    "BuildDate": "2022-02-03T20:41:00Z",
    "GitCommit": "78d749ec88bea5a5e851fc48b9404fdf067192b6",
    "GitTreeState": "clean",
    "GoVersion": "go1.16.11",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v4.2.0 2021-06-30T22:49:26Z",
    "HelmVersion": "v3.7.1+g1d11fcb",
    "KubectlVersion": "v0.22.2",
    "JsonnetVersion": "v0.17.0"
}
```
